include/LightGBM/config.h

```diff
@@ -1105,21 +1111,24 @@ struct Config {
   #pragma region GPU Parameters
   #endif  // __NVCC__
 
+  // desc = used only with ``gpu`` device type
   // desc = OpenCL platform ID. Usually each GPU vendor exposes one OpenCL platform
   // desc = ``-1`` means the system-wide default platform
   // desc = **Note**: refer to `GPU Targets <./GPU-Targets.rst#query-opencl-devices-in-your-system>`__ for more details
   int gpu_platform_id = -1;
 
-  // desc = OpenCL device ID in the specified platform. Each GPU in the selected platform has a unique device ID
+  // desc = OpenCL device ID in the specified platform or CUDA device ID. Each GPU in the selected platform has a unique device ID
   // desc = ``-1`` means the default device in the selected platform
   // desc = **Note**: refer to `GPU Targets <./GPU-Targets.rst#query-opencl-devices-in-your-system>`__ for more details
   int gpu_device_id = -1;
 
+  // desc = used only with ``gpu`` device type
   // desc = set this to ``true`` to use double precision math on GPU (by default single precision is used)
   // desc = **Note**: can be used only in OpenCL implementation, in CUDA implementation only double precision is currently supported
   bool gpu_use_dp = false;
 
   // check = >0
+  // desc = used only with ``cuda`` device type
   // desc = number of GPUs
   // desc = **Note**: can be used only in CUDA implementation
```

Comment:
similar to the comment on `gpu_use_dp`, what do you think about combining two lines here? So we'd have:

```text
  // desc = **Note**: can be used only in CUDA implementation (``device_type="cuda"``)
```